src/rules.c

```diff
@@ -14,66 +14,106 @@
 
 GSList *rules = NULL;
 
+#define APPLY(nprop, rprop, defval) \
```

Comment:
Personally, I would scope this macro just for `rule_apply`. This way, it's also more obvious what `r`, `n` and `save`.

You could also move the initial condition to the macro as it always seems to be
```c
if (r->rprop != defval)
```

Do we have any cases where `nprop` and `rprop` differ?

Reply:
> Do we have any cases where `nprop` and `rprop` differ?

unfortunately yes. But I made a secondary macro to fix this